v2/internal/parse_test.go

```diff
@@ -28,23 +33,180 @@ func TestParseMessageFileBytes(t *testing.T) {
+			file: `{"outer": {"nested": {"inner": "value"}}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
+		{
+			name: "multiple nested test",
+			file: `{"nested": {"hello": "world", "bye": "all"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "nested.hello",
+					Other: "world",
+				}, {
+					ID:    "nested.bye",
+					Other: "all",
+				}},
+			},
+		},
+		{
+			name: "YAML nested test",
+			file: `
+outer:
+    nested:
+        inner: "value"`,
+			path:           "en.yaml",
+			unmarshalFuncs: map[string]UnmarshalFunc{"yaml": yaml.Unmarshal},
+			messageFile: &MessageFile{
+				Path:   "en.yaml",
+				Tag:    language.English,
+				Format: "yaml",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
 	}
 	for _, testCase := range testCases {
 		actual, err := ParseMessageFileBytes([]byte(testCase.file), testCase.path, testCase.unmarshalFuncs)
-		if err != testCase.err {
-			t.Fatalf("expected error %#v; got %#v", testCase.err, err)
+		if (err == nil && testCase.err != nil) ||
+			(err != nil && testCase.err == nil) ||
+			(err != nil && testCase.err != nil && err.Error() != testCase.err.Error()) {
+			t.Errorf("%s failed: expected error %#v; got %#v", testCase.name, testCase.err, err)
+			continue
+		}
+		if actual == nil {
+			continue
 		}
 		if actual.Path != testCase.messageFile.Path {
-			t.Fatalf("expected path %q; got %q", testCase.messageFile.Path, actual.Path)
+			t.Errorf("%s failed: expected path %q; got %q", testCase.name, testCase.messageFile.Path, actual.Path)
+			continue
 		}
 		if actual.Tag != testCase.messageFile.Tag {
-			t.Fatalf("expected tag %q; got %q", testCase.messageFile.Tag, actual.Tag)
+			t.Errorf("%s failed: expected tag %q; got %q", testCase.name, testCase.messageFile.Tag, actual.Tag)
+			continue
 		}
 		if actual.Format != testCase.messageFile.Format {
-			t.Fatalf("expected format %q; got %q", testCase.messageFile.Format, actual.Format)
+			t.Errorf("%s failed: expected format %q; got %q", testCase.name, testCase.messageFile.Format, actual.Format)
+			continue
+		}
+		if !equalMessages(actual.Messages, testCase.messageFile.Messages) {
+			t.Errorf("%s failed: expected %#v; got %#v", testCase.name, testCase.messageFile.Messages, actual.Messages)
+			continue
 		}
-		if !reflect.DeepEqual(actual.Messages, testCase.messageFile.Messages) {
-			t.Fatalf("expected %#v; got %#v", testCase.messageFile.Messages, actual.Messages)
+	}
+}
+
+// equalMessages compares two slices of messages, ignoring private fields and order.
+// Sorts both input slices, which are therefore modified by this function.
+func equalMessages(m1, m2 []*Message) bool {
```

Comment:
I don't think this function is necessary anymore?

Reply:
It is, for the `"multiple nested test"` subtest: this test returns 2 messages nested under the same key.
Since map keys enumeration is random in Go, in theory these 2 messages can be created in any order (in practice, first line is created first 90% of the time though), which makes the test fail randomly if we do not make order-insensitive comparison.

It is required only in `TestParseMessageFileBytes()`, so I moved it in `parse_test.go` as an unexported function, but we need it.